AdvancedDLSupport/ImplementationGenerators/Wrappers/SpanMarshallingWrapper.cs

```diff
@@ -98,13 +131,50 @@ public override IntrospectiveMethodInfo GeneratePassthroughDefinition(PipelineWo
             );
         }
 
+        /// <inheritdoc/>
+        public override void EmitPrologue(ILGenerator il, PipelineWorkUnit<IntrospectiveMethodInfo> workUnit)
+        {
+            var definition = workUnit.Definition;
+
+            var parameterTypes = definition.ParameterTypes;
+
+            il.EmitLoadArgument(0);
+
+            for (short i = 1; i <= parameterTypes.Count; ++i)
+            {
+                var paramType = parameterTypes[i - 1];
+
+                if (IsSpanType(paramType))
+                {
+                    Debug.Assert(paramType.GenericTypeArguments.Length == 1, "Span Type does not the correct number of generic parameters (1), CLR/BCL bug?");

```

Comment:
What's this?

Reply:
An unnecessary check by the looks of things, ensuring that you can't pass `Span<T1, T2>` which doesn't exist